app/Jobs/ImportRockTheVotePosts.php

```diff
@@ -0,0 +1,459 @@
+                        $postDetails = $this->extractDetails($record);
+
+                        $postData = [
+                            'campaign_id' => (int) $referralCodeValues['campaign_id'],
+                            'campaign_run_id' => (int) $referralCodeValues['campaign_run_id'],
+                            'northstar_id' => $user->id,
+                            'type' => 'voter-reg',
+                            'action' => $rtvCreatedAtMonth . '-rockthevote',
+                            'status' => $this->translateStatus($record['Status'], $record['Finish with State']),
+                            'source' => 'rock-the-vote',
+                            'source_details' => $sourceDetails,
+                            'details' => $postDetails,
+                        ];
+
+                        try {
+                            $post = $rogue->createPost($postData);
+
+                            if ($post['data']) {
+                                $this->stats['countPostCreated']++;
+                            }
+                        } catch (\Exception $e) {
+                            info('There was an error storing the post', [
+                                'Error' => $e->getMessage(),
+                            ]);
+                        }
+                    } else {
+                        $newStatus = $this->translateStatus($record['Status'], $record['Finish with State']);
+                        $statusShouldChange = $this->updateStatus($post['data'][0]['status'], $newStatus);
+
+                        if ($statusShouldChange) {
+                            try {
+                                $rogue->updatePost($post['data'][0]['id'], ['status' => $statusShouldChange]);
+                            } catch (\Exception $e) {
+                                info('There was an error updating the post for: ' . $record['Email address'], [
+                                    'Error' => $e->getMessage(),
+                                ]);
+                            }
+                        }
+                    }
+                }
+                $this->stats['countProcessed']++;
+            } else {
+                $this->stats['countScrubbed']++;
+            }
+
+            event(new LogProgress('', 'progress', ($offset / $this->totalRecords) * 100));
+        }
+
+        event(new LogProgress('Done!', 'general'));
+
+        Stat::create([
+            'filename' => $this->filepath,
+            'total_records' => $this->stats['totalRecords'],
+            'stats' => json_encode($this->stats),
+        ]);
+
+    }
+
+    /**
+     * Initiate the stat counters.
+     *
+     * @return array
+     */
+    private function statsInit()
+    {
+        return [
+            'totalRecords' => 0,
+            'countScrubbed' => 0,
+            'countProcessed' => 0,
+            'countPostCreated' => 0,
+            'countUserAccountsCreated' => 0,
+        ];
+    }
+
```

Comment:
same with this! Is this possible to pull out to a helper?

Reply:
good call!!